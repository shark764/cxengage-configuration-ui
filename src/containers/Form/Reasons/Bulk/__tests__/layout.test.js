/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import ReasonsBulkActionsForm from '../layout';
import { fromJS } from 'immutable';

describe('<ReasonsBulkActionsForm />', () => {
  it('renders reasons bulk actions form', () => {
    const rendered = shallow(
      <ReasonsBulkActionsForm
        entityName={`reasons`}
        userHasUpdatePermission={true}
        userHasSharePermission={false}
        initialValues={fromJS({ shared: false })}
        sharedIsChecked={true}
        isSaving={true}
        isBulkUpdating={true}
        change={() => {}}
        handleSubmit={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
