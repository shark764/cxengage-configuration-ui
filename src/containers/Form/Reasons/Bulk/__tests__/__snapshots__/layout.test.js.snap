// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReasonsBulkActionsForm /> renders reasons bulk actions form 1`] = `
<form
  onSubmit={[Function]}
>
  <styled.div>
    <styled.span>
      <span>
        Enable/Disable 
        Reasons
      </span>
      <Styled(Toggle)
        onChange={[Function]}
        value={false}
      />
    </styled.span>
  </styled.div>
  <styled.div>
    <styled.span>
      <span>
        Share 
        Reasons
      </span>
      <ConfirmationWrapper
        data-automation="toggleSharedConfirmationWrapper"
        mainText="Setting shared to 'enabled' cannot be reverted."
        secondaryText="Are you sure you want to continue?"
      >
        <Styled(ToggleField)
          data-automation="sharedToggle"
          disabled={true}
          label=""
          name="shared"
        />
      </ConfirmationWrapper>
    </styled.span>
    <styled.p>
      <strong>
        Note:
      </strong>
       This action cannot be undone.
    </styled.p>
  </styled.div>
</form>
`;
